Mage/src/main/java/mage/abilities/common/CommittedCrimeTriggeredAbility.java

```diff
@@ -0,0 +1,128 @@
+        super(ability);
+    }
+
+    @Override
+    public CommittedCrimeTriggeredAbility copy() {
+        return new CommittedCrimeTriggeredAbility(this);
+    }
+
+    @Override
+    public boolean checkEventType(GameEvent event, Game game) {
+        switch (event.getType()) {
+            case SPELL_CAST:
+            case ACTIVATED_ABILITY:
+            case TRIGGERED_ABILITY:
+                return true;
+            default:
+                return false;
+        }
+    }
+
+    @Override
+    public boolean checkTrigger(GameEvent event, Game game) {
+        return isControlledBy(getCriminal(event, game));
+    }
+
+    public static UUID getCriminal(GameEvent event, Game game) {
+        UUID controllerId;
+        Ability ability;
+        switch (event.getType()) {
+            case SPELL_CAST:
+                Spell spell = game.getSpell(event.getTargetId());
+                if (spell == null) {
+                    return null;
+                }
+                controllerId = spell.getControllerId();
+                ability = spell.getSpellAbility();
+                break;
+            case ACTIVATED_ABILITY:
+            case TRIGGERED_ABILITY:
+                StackObject stackObject = game.getStack().getStackObject(event.getTargetId());
+                if (stackObject == null) {
+                    return null;
+                }
+                controllerId = stackObject.getControllerId();
+                ability = stackObject.getStackAbility();
+                break;
+            default:
+                return null;
+        }
+        if (controllerId == null || ability == null) {
+            return null;
+        }
+        Set<UUID> opponents = game.getOpponents(controllerId);
+        Set<UUID> targets = ability
+                .getModes()
+                .values()
+                .stream()
+                .map(Mode::getTargets)
+                .flatMap(Collection::stream)
+                .map(Target::getTargets)
+                .flatMap(Collection::stream)
+                .collect(Collectors.toSet());
+        // an opponent
+        if (targets
+                .stream()
+                .anyMatch(opponents::contains)
+                // an opponent's permanent
+                || targets
+                .stream()
+                .map(game::getPermanent)
+                .filter(Objects::nonNull)
+                .map(Controllable::getControllerId)
+                .anyMatch(opponents::contains)
+                // an opponent's spell
+                || targets
+                .stream()
+                .map(game.getStack()::getStackObject)
+                .filter(Objects::nonNull)
+                .map(Controllable::getControllerId)
+                .anyMatch(opponents::contains)
+                // a card in an opponent's graveyard
+                || targets
+                .stream()
+                .filter(uuid -> Zone.GRAVEYARD.match(game.getState().getZone(uuid)))
```

Comment:
For info: current wiki page contains talking about any zones, not graveyard only:
https://mtg.fandom.com/wiki/Committing_a_crime
![shot_240327_222012](https://github.com/magefree/mage/assets/8344157/26b27ea9-497f-4c18-93a3-f585acbc4b62)

[All scryfall's cards](https://scryfall.com/search?q=oracle%3A%22commit+a+crime%22) contains graveyard only. So maybe wiki is outdated. Or maybe other cards will come. Keep in mind.

Reply:
I had it that way before but the [mechanics article](https://magic.wizards.com/en/news/feature/outlaws-of-thunder-junction-mechanics) says it's just graveyard, I think there was a miscommunication before.